File: frontend/src/types/thread.ts
// Mirrors backend/app/schemas/thread.py — keep in sync.
//
// Contract for GET /api/threads (bearer-protected):
//   { "threads": Thread[] }
// Threads are returned newest-first; the client groups them by day.

export type ThreadStatus =
  | "idle"
  | "in_progress"
  | "awaiting_input"
  | "draft";

export interface Thread {
  id: string;
  title: string;
  /** Optional 2-line preview of the latest exchange. */
  preview?: string | null;
  status: ThreadStatus;
  /** ISO-8601 timestamp of the last activity. */
  updatedAt: string;
  /** Number of turns so far, shown in the card footer when present. */
  turnCount?: number | null;
}

export interface ThreadListResponse {
  threads: Thread[];
}

/** A thread's footer status drives the pulse pip in the sidebar. */
export const ACTIVE_STATUSES: ReadonlySet<ThreadStatus> = new Set([
  "in_progress",
  "awaiting_input",
]);

File: backend/app/schemas/thread.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel

ThreadStatus = Literal["idle", "in_progress", "awaiting_input", "draft"]


class Thread(BaseModel):
    """A single agent conversation thread, as shown in the sidebar history."""

    id: str
    title: str
    preview: str | None = None
    status: ThreadStatus = "idle"
    updatedAt: datetime
    turnCount: int | None = None


class ThreadListResponse(BaseModel):
    """Response for GET /api/threads — threads are returned newest-first."""

    threads: list[Thread]
